mlflow/server/js/src/experiment-tracking/components/MetricsPlotPanel.js

```diff
@@ -436,6 +439,35 @@ export class MetricsPlotPanel extends React.Component {
     };
     this.updateUrlState({ layout: mergedLayout, lastLinearYAxisRange });
   };
+  
+  convertJsonToCsv = () => {
+    const metrics = this.getMetrics();
+    const selectedMetricKeys = this.getUrlState().selectedMetricKeys;
+    console.log("selected metrics: " + selectedMetricKeys);
+    const heading = Object.keys(metrics[0].history[0]).join(",") + "\n"
+    const metricValues = selectedMetricKeys.map(
+      (metricKey) => {
+        return metrics
+          .filter((metric) => metric.metricKey === metricKey)
+          .map((metric) => {
+            return metric.history
+              .map((metricValue) => {
+                return Object.values(metricValue);
+              }).join("\n");
+          });
+      }).join("\n");
+
+    return `${heading}${metricValues}`;
+  };
+
+  onDownloadCsv = () => {
+    const csv = this.convertJsonToCsv();
+    console.log("CSV: " + csv)
+    const blob = new Blob([csv], { type: 'application/csv;charset=utf-8' });
+    console.log("Blob: " + blob)
+    const selectedRunUuid = this.props.runUuids[0]
```

Comment:
The metric plot may contain multiple runs. Does the current implementation work in such case?

Reply:
@harupy can you point me to how displaying metrics from multiple runs? on the screen I worked with there's only one, always, and possibly more than 1 metric:

<img width="427" alt="Screenshot 2022-06-22 at 18 30 27" src="https://user-images.githubusercontent.com/97888102/175087220-63288e15-9b87-4c4d-978d-f251a3637999.png">

Reply:
@harupy Metrics comparison across different runs is part of CompareRun*.js components. MetricsPlotPanel is not referenced or used in this component, nor in any other except the following (happy to be corrected in case I'm missing something):

<img width="657" alt="Screenshot 2022-06-23 at 03 40 01" src="https://user-images.githubusercontent.com/97888102/175189025-15ffe930-ca8c-4f9b-8d66-06965ae447ca.png">

Agree it would make sense to extend this functionality to CompareRuns as well but for the sake of sticking to the original scope of this PR I would suggest we tackle that as part of a separate one.

Reply:
here's how it looks now - metrics page for multiple runs:

<img width="1676" alt="Screenshot 2022-06-23 at 05 05 38" src="https://user-images.githubusercontent.com/97888102/175198524-a2ac591d-11df-43ab-901f-d83b4b9ea363.png">

resulting CSV file:

<img width="475" alt="Screenshot 2022-06-23 at 05 06 45" src="https://user-images.githubusercontent.com/97888102/175198688-cf48a7bb-8636-4538-ae00-93c6e363b8ca.png">

Reply:
also changed the file name to "metric_values.csv" since it can contain multiple runs